modules/fx/arrow-fx/src/main/kotlin/arrow/fx/typeclasses/Concurrent.kt

```diff
@@ -970,10 +971,10 @@ interface Concurrent<F> : Async<F> {
    * ```
    **/
   fun <A> Kind<F, A>.waitFor(duration: Duration): Kind<F, A> =
-    dispatchers().default().raceN(this, sleep(duration)).flatMap {
+    EmptyCoroutineContext.raceN(sleep(duration), this).flatMap {
```

Comment:
Without this change the `Eq` instance below  makes the tests hang forever.

```
IO.applicative().mapN(a.attempt(), b.attempt()) { (a, b) -> a.eqv(b) }
  .waitFor(timeout)
  .unsafeRunSync()
```

This change removes a context switch, is guaranteed to fork due to `sleep` being the first argument, allowing `this` to run in its original context. So  `waitFor` now *does not* fork `this` on the `default` dispatcher anymore, which is more desirable IMO.